pkg/systemd/logs.js

```diff
@@ -863,6 +864,18 @@ $(function() {
         $("body").show();
     }
 
+    function fit_filters() {
+        const filters = $("#journal .content-header-extra .filter");
+        if ($(filters[filters.length - 1]).offset().top > 20) {
```

Comment:
Well, the thing is that css screen sizes won't work here. We need to be absolutely sure when things started to wrap (because we want to keep the date sticky to the top and having two rows would break that).
We can set 'max-width' to every single element, but that will break things when translations are used. I think we are better off to just check if the last element broke to another line and then just apply the logic, than to try to guess screen sizes and when it might break in which languages...
The hardcoded `20` just means that the element is further than 20px from the top of the parent - which happens only when it wraps. (margin is 10px)

Reply:
I guess I should look at this. It really has bad "[code smell](https://en.wikipedia.org/wiki/Code_smell)", as it's doing presentation-related logic in JS instead of using CSS.

You shouldn't have to:
- use a string of selectors in the JS
- care about pixel values (which are easily messed up with browser zoom / text zoom — and browser zoom is very common; recent Firefox releases even support global zooming across all sites).

Reply:
Note: Even changing the OS to large fonts — a setting completely independent of the browser — will impact things in ways you won't expect.

(You may need to restart a browser after enabling "large text" in GNOME's settings, or something similar on other desktops. Firefox doesn't change the fonts on the fly, but Chrome seems to.)

Reply:
This has been changed all together.